controllers/articles-controller.js

```diff
@@ -40,4 +40,66 @@ exports.getCommentsForArticle = (req, res, next) => {
     .catch((err) => {
         next(err)
     })
-};
\ No newline at end of file
+};
+
+exports.postComment = (req, res, next) => {
+    const id = req.params.article_id;
+    const body = req.body;
+
+    if (!checkCommentStructure(body)) next({ 
+        statusCode: 400, msg: 'Bad request! Missing or incorrect key name(s)' 
+    });    
+
+    checkUserExists(body.username)
+    .then((userExists) => {
+        if (!userExists) {
+            next({statusCode: 401, msg: 'You need to have an account to post comments' });

```

Comment:
Just be careful with error messages aimed at users. Normally, it will be another developer seeing these messages trying to connect it to a front-end.

Reply:
It makes sense therefore to change the msg to something like 'The user needs to (have an account || log in) to post comments' ?

Reply:
You would likely expect something along the lines of 'User not found'